tests/native_tests/language.hy

```diff
@@ -1037,20 +993,6 @@
   (assert (is (xor 0 False) False)))
 
 
-(defn test-if-return-branching []
```

Comment:
How much of this test was actually do to `let`? It was added in [this bugfix commit](https://github.com/hylang/hy/commit/5336b2f71ace09ab3df9af56238eb16289f0aafa), yet the bugfix didn't seem to have anything to do with let. Was this fixing another bug that just *happened* to be easy to reproduce using `let`?